### PR TITLE
feat(dependency-track): support parent ID for autocreate projects (#534)

### LINKED ISSUE #544: feat: revisit requirements to leave org
Currently an user can not leave an org if it's the one set as `current`. We should evaluate if this is a requirement that can be dropped.

### PR DESCRIPTION
**Pull request by sedan07**
For now just supporting parent ID, not name and version. Although I do think it would be useful to support project version as well as parent name and version, this way a workflow could be shared by several projects if the parent name/version could be read from annotations.

No testing for the HTTP submission to dependency-track yet, although I'm not sure the best way to approach that in Go.

closes #544

### CODE REVIEW

File: app/controlplane/plugins/core/dependency-track/v1/extension.go [hunk 1]
```diff
@@ -46,6 +47,17 @@ type attachmentRequest struct {
 	// Either one or the other
 	ProjectID   string `json:"projectID,omitempty" jsonschema:"oneof_required=projectID,minLength=1,description=The ID of the existing project to send the SBOMs to"`
 	ProjectName string `json:"projectName,omitempty" jsonschema:"oneof_required=projectName,minLength=1,description=The name of the project to create and send the SBOMs to"`
+
+	ParentID string `json:"parentID,omitempty" jsonschema:"minLength=1,description=ID of parent project to create a new project under"`
+}
+
+// invopop/jsonschema doesn't appear to support dependentRequired through reflection
+func (x attachmentRequest) JSONSchemaExtend(schema *jsonschema.Schema) {
```
[the remark this answers is not in the document]
Comment (sedan07) [reply app/controlplane/plugins/core/dependency-track/v1/extension.go:56]: Good idea, i've added a brief explanation.

File: app/controlplane/plugins/core/dependency-track/v1/extension.go [hunk 2]
```diff
@@ -57,6 +69,7 @@ type registrationConfig struct {
 type attachmentConfig struct {
 	ProjectID   string `json:"projectId"`
 	ProjectName string `json:"projectName"`
+	ParentID    string `json:"parentID"`
```
[the remark this answers is not in the document]
Comment (sedan07) [reply app/controlplane/plugins/core/dependency-track/v1/extension.go]: Makes sense. Updated.

**Review by danlishka — APPROVED** [review]
LGTM. I won't have to move DT projects around manually anymore. Thank you!

**Comment by migmartri** [comment]
Merged! Congrats for your first contribution to Chainloop!

**Pull request merged**